### PR TITLE
Stickiness improvement and filter refactor

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- adds interval handling to stickiness
- refactors stickiness to use specific filter and process parameters within filter
- add a bunch of tests for stickiness
- moves tests and person handling under test_stickiness and out of test_action_people
- adds team_id argument to `GET_EARLIEST_TIMESTAMP` ch query

**New tests and code movement account for big line diff**

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: posthog/models/filters/stickiness_filter.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            }
+        elif not data:
+            raise ValueError("You need to define either a data dict or a request")
+
+        team: Optional[Team] = kwargs.get("team", None)
+        if not team:
+            raise ValueError("Team must be provided to stickiness filter")
+
+        if self._date_from == "all":
+            return (
```
Comment (macobo) [inline posthog/models/filters/stickiness_filter.py:40]: Why are we returning data in a constructor? :confused:

Comment (EDsCODE) [reply posthog/models/filters/stickiness_filter.py:40]: fantastic question..was pasting over earliest date logic

Comment (EDsCODE) [reply posthog/models/filters/stickiness_filter.py:40]: thanks for the catch—haven't finished writing all tests yet. stickiness was severely lacking tests for most filtering

**Comment by timgl** [comment]
Merge conflicts!

**Comment by EDsCODE** [comment]
fixed!

**Pull request merged**